app_dart/lib/src/service/config.dart

```diff
@@ -9,21 +9,25 @@ import 'package:cocoon_server/generate_github_jws.dart';
 import 'package:cocoon_server/logging.dart';
 import 'package:cocoon_server/secret_manager.dart';
 import 'package:github/github.dart' as gh;
-import 'package:graphql/client.dart';
+import 'package:graphql/client.dart' hide JsonSerializable;
 import 'package:http/http.dart' as http;
+import 'package:json_annotation/json_annotation.dart';
 import 'package:meta/meta.dart';
 import 'package:retry/retry.dart';
 
 import '../../cocoon_service.dart';
 import 'github_service.dart';
 import 'luci_build_service/cipd_version.dart';
 
+part 'config.g.dart';
+
 /// Name of the default git branch.
 const String kDefaultBranchName = 'master';
 
 interface class Config {
   /// Creates and returns a [Config] instance.
-  Config(this._cache, this._secrets);
+  Config(this._cache, this._secrets, {DynamicConfig? dynamicConfig})
```

Comment:
It would be nice to not accidentally initialize this to the default state unless that is expected.

In other words, something like:
```dart
Config(this._cache, this._secrets, {required DynamicConfig initialConfig}) : dynamicConfig = initialConfig;
Config.useRepoForInitialConfig(this._cache, this._secrets) : dynamicConfig = DynamicConfig.fromJson({});
```

Or if you prefer doing:
```dart
// Make it required to explicitly write `config: null` if we don't want to use the fetched config.
Config(this._cache, this._secrets, {required DynamicConfig? config})
```

I personally like (1) but (2) is also fine.

Reply:
I'll just fix FakeConfig - it should be required to have dynamic config first.